### PR TITLE
feature/AKD-145: Added 'VM Approved Extensions' plugin and test cases

### PR DESCRIPTION
**Pull request by ali-imran7**

### CODE REVIEW

File: plugins/azure/virtualmachines/approvedVmExtension.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+var async = require('async');
+
+var helpers = require('../../../helpers/azure/');
+
+module.exports = {
+    title: 'VM Approved Extensions',
+    category: 'Virtual Machines',
+    description: 'Ensures that approved virtual machine extensions are installed.',
+    more_info: 'Extensions are small applications that provide post-deployment configuration and automation on Azure VMs. VMs installed should be approved by the organization to meet the organizational security requirements.',
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/approvedVmExtension.js:9]:
```suggestion
    more_info: 'Extensions are small applications that provide post-deployment configuration and automation on Azure VMs. Extensions installed should be approved by the organization to meet the organizational security requirements.',
```

File: plugins/azure/virtualmachines/approvedVmExtension.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                }
+                
+                if (!virtualMachineExtensions.data.length) {
+                    helpers.addResult(results, 0, 'No VM Extensions found', location, virtualMachine.id);
+                    return scb();
+                }      
+                
+                virtualMachineExtensions.data.forEach(function(virtualMachineExtension) {
+                    if (config.approvedExtensions.length) {
+                        let extensionsList = config.approvedExtensions.split(',');
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/approvedVmExtension.js:63]: Move this on top

File: plugins/azure/virtualmachines/approvedVmExtension.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                        let found = extensionsList.some(extension => extension.trim() === virtualMachineExtension.name);
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/approvedVmExtension.js:64]: Use find

File: plugins/azure/virtualmachines/approvedVmExtension.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                        if (found) {
```
Comment (AkhtarAmir) [inline plugins/azure/virtualmachines/approvedVmExtension.js:65]: Move this check outside foreach loop

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/virtualmachines/approvedVmExtension.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    run: function(cache, settings, callback) {
+        var results = [];
+        var source = {};
+        var locations = helpers.locations(settings.govcloud);
+
+        const config = {
+            approvedExtensions: (settings.vm_approved_extensions || this.settings.vm_approved_extensions.default)
+        };
+
```
Comment (giorod3) [inline plugins/azure/virtualmachines/approvedVmExtension.js:30]: this plugin should be opt-in and skipped here if no setting exists

**Review by giorod3 — CHANGES_REQUESTED** [review]

**Review by giorod3 — APPROVED** [review]

**Pull request merged**